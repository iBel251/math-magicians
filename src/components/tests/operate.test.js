import React from 'react';
import renderer from 'react-test-renderer';
import operate from '../../logic/operate';

describe('Testin for all ', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<operate />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Addition test', () => {
    expect(operate(12, 20, '+')).toEqual('32');
  });
  test('Subtraction test', () => {
    expect(operate(32, 20, '-')).toEqual('12');
  });
  test('Multiply test', () => {
    expect(operate(10, 20, 'x')).toEqual('200');
  });
  test('Division test', () => {
    expect(operate(20, 2, 'รท')).toEqual('10');
  });
});
