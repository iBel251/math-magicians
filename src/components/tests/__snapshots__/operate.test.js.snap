// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testin for all  renders correctly 1`] = `<operate />`;

exports[`renders correctly 1`] = `<operate />`;
